SocialNetwork/SocialNetwork.Core/Services/UserService.cs

```diff
@@ -1,36 +1,67 @@
 ﻿using SocialNetwork.Core.Interfaces;
 using SocialNetwork.Core.Models;
-
+using SocialNetwork.Core.Helpers;
+using System.Text.RegularExpressions;
+using Microsoft.EntityFrameworkCore;
 namespace SocialNetwork.Core.Services;
-
 public class UserService : IUserService
 {
-    private readonly IRepository _repository;
-
+    private readonly IRepository repository;
     public UserService(IRepository repository)
     {
-        _repository = repository;
+        this.repository = repository;
     }
-
-    public IEnumerable<User> GetAllUsers(int skip = 0, int take = 20)
+    public IEnumerable<User> GetUsers() 
     {
-        return _repository.GetAll<User>()
-            .Skip(skip)
-            .Take(take);
+        return repository.GetAll<User>();
     }
-
-    public async Task<User> AddUser(User user)
+    public async Task<User> GetUserById(int id)
     {
-        throw new NotImplementedException();
+        var user = await repository.GetById<User>(id);
+        if (user == null)
+            throw new ArgumentException("User not found"); // TODO own types of exceptions
+        return user;
     }
-
-    public Task<User> UpdateUser(User user)
+    public async Task<User?> GetUserByName(string name)
     {
-        throw new NotImplementedException();
+        var users = repository.GetAll<User>();
+        var user = await users.FirstOrDefaultAsync(u => u.Nickname == name);
```

Comment:
Consider using `SingleOrDefaultAsync()`, if there is two users with same nickname by some mistake we will be able to investigate it